src/libraries/System.Security.Cryptography/tests/X509Certificates/CertTests.cs

```diff
@@ -571,6 +571,86 @@ public static void SignatureAlgorithmOidReadableForGostCertificate()
             }
         }
 
+        [ConditionalFact(typeof(PlatformSupport), nameof(PlatformSupport.PlatformCryptoProviderFunctional))]
+        [OuterLoop("Hardware backed key generation takes several seconds.")]
+        public static void CreateCertificate_MicrosoftPlatformCryptoProvider_EcdsaKey()
+        {
+             CngKey key = null;
+
+            try
+            {
+                CngKeyCreationParameters cngCreationParameters = new CngKeyCreationParameters
+                {
+                    Provider = CngProvider.MicrosoftPlatformCryptoProvider,
+                    KeyCreationOptions = CngKeyCreationOptions.OverwriteExistingKey,
+                };
+
+                key = CngKey.Create(
```

Comment:
nit: the creation of `CngKeyCreationParameters` could be moved out of try/finally block:

```suggestion
            CngKeyCreationParameters cngCreationParameters = new CngKeyCreationParameters
            {
                Provider = CngProvider.MicrosoftPlatformCryptoProvider,
                KeyCreationOptions = CngKeyCreationOptions.OverwriteExistingKey,
            };

            CngKey key = null;

            try
            {
                key = CngKey.Create(
```

Reply:
Thanks for the suggestion. I ended up reworking this in to a helper to avoid copying and pasting the same try/finally over and over.